android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -760,6 +760,39 @@ class GeckoEngineSession(
         })
     }
 
+    /**
+     * See [EngineSession.requestCreateProductAnalysis]
+     */
+    @Suppress("ComplexCondition")
+    override fun requestCreateProductAnalysis(
+        url: String,
+        onResult: (String) -> Unit,
+        onException: (Throwable) -> Unit,
+    ) {
+        geckoSession.requestCreateAnalysis(url).then({
+                response ->
+            if (response == null) {
```

Comment:
❓ When can response be null?

Reply:
Removed null check, GV should return NonNull value